src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -0,0 +1,99 @@
+package com.hibob.academy.dao
+
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.RecordMapper
+import org.jooq.impl.DSL
+import org.springframework.stereotype.Component
+import java.math.BigInteger
+import java.time.LocalDateTime
+import java.sql.Date
+
+data class PetCreationRequest(
+    val name: String,
+    val type: String,
+    val dateOfArivval: Date,
+    val companyId: Long,
+    val ownerId: Long
+)
+
+data class Pet(
+    val id: Long,
+    val name: String,
+    val type: String,
+    val dateOfArivval: Date,
+    val companyId: Long,
+    val ownerId: Long
+)
+
+
+@Component
+class PetDao(private val sql: DSLContext) {
+    private val table = PetTable.instance
+    private val ownerTable = OwnerTable.instance
+    private val ownerDao = OwnerDao(sql)
```

Comment:
Why do we need the ownerDao in the petDao?

Reply:
i need for use the ownermapper - now i put the owner mapper static - companion object - is it better ?

Reply:
First of all if you want to use the ownerDao you need to inject it.
Im not sure that this is the right way to do.
Maybe move the join function to other ownerDao?
Let me know what you decide and why.

Reply:
I moved the function to the ownerDao -and do the left join from there.  I did it because now in the petDao i dont need the ownerDao , the same about the ownerDao .